lib/ui/widgets/basewidget.js

```diff
@@ -79,4 +78,53 @@ BaseWidget.prototype.getHandledSources = function() {
   }];
 }
 
+BaseWidget.prototype.pushProps = function(widget, valueStream) {
+  throw new Error('pushProps not implemented');
+}
+
+BaseWidget.prototype.updateFromOnlineMeta = function(widget) {
+  var valueStream = widget.streamBundle.getStreamForSourcePath(widget.options.sourceId, widget.options.path);
+  widget.updateFromOnlineMeta = 1;
+  signalkUnits = new signalkClient(widget.instrumentPanel.host.split(':')[0], widget.instrumentPanel.host.split(':')[1], widget.instrumentPanel.useSSL);
+  signalkUnits.apiGet('/vessels/self/' + widget.options.path.replace(/\./g, '/') + '/meta/')
+  .then(function (response) {
+//    console.log('updateFromOnlineMeta:GET => '+response.status+':'+response.req.url+':'+response.text);
+    var meta = JSON.parse(response.text);
```

Comment:
something like this would be more idiomatic:
```
fetch('./data.json').then(response => {
  return response.json();
}).then(data => {
  // Work with JSON data here
  console.log(data);
}).catch(err => {
  // Do something for an error here
});
```

Reply:
OK, used 'fetch' and removed signalkClient.